### PR TITLE
Move `query_archetype_with_history` to make it more globally available

### PR DESCRIPTION
**Pull request by abey79**
### What

Small refactor discussed with @Wumpf to make `query_archetype_with_history` consistent with `latest_at_with_blueprint_resolved_data`.

Needed by #6561 

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)
* [x] I have tested the web demo (if applicable):
  * Using examples from latest `main` build: [rerun.io/viewer](https://rerun.io/viewer/pr/7863?manifest_url=https://app.rerun.io/version/main/examples_manifest.json)
  * Using full set of examples from `nightly` build: [rerun.io/viewer](https://rerun.io/viewer/pr/7863?manifest_url=https://app.rerun.io/version/nightly/examples_manifest.json)
* [x] The PR title and labels are set such as to maximize their usefulness for the next release's CHANGELOG
* [x] If applicable, add a new check to the [release checklist](https://github.com/rerun-io/rerun/blob/main/tests/python/release_checklist)!
* [x] If have noted any breaking changes to the log API in `CHANGELOG.md` and the migration guide

- [PR Build Summary](https://build.rerun.io/pr/7863)
- [Recent benchmark results](https://build.rerun.io/graphs/crates.html)
- [Wasm size tracking](https://build.rerun.io/graphs/sizes.html)

To run all checks from `main`, comment on the PR with `@rerun-bot full-check`.

### CODE REVIEW

File: crates/viewer/re_space_view_spatial/src/visualizers/utilities/entity_iterator.rs [hunk 1]
```diff
@@ -178,12 +134,11 @@ where
             space_view_class_identifier: view_ctx.space_view_class_identifier(),
         };
 
-        let results = query_archetype_with_history::<A>(
+        let results = data_result.query_archetype_with_history::<A>(
             ctx,
             &query.timeline,
             query.latest_at,
             data_result.query_range(),
```
Comment (Wumpf) [inline crates/viewer/re_space_view_spatial/src/visualizers/utilities/entity_iterator.rs:141]: Do we ever pass anything but the data_result's query range in here? If not then we can drop that argument :)

Also if I'm not mistaken, `re_space_view` knows about the `ViewQuery` type we're using here, so we might just pass a reference to that instead.


Disregard that if there's friction, I don't want you to have to expand your tangent here more than necessary.

**Review by Wumpf — APPROVED** [review]
neat!

File: crates/viewer/re_space_view_spatial/src/visualizers/utilities/entity_iterator.rs (hunk 1 above)
Comment (abey79) [reply crates/viewer/re_space_view_spatial/src/visualizers/utilities/entity_iterator.rs:141]: 👀 (maybe tomorrow)

Comment (abey79) [reply crates/viewer/re_space_view_spatial/src/visualizers/utilities/entity_iterator.rs]: All done, very nice improvement.

**Pull request merged**